akka-remote/src/main/scala/akka/remote/artery/RemoteInstrument.scala

```diff
@@ -16,177 +20,272 @@ import akka.util.{ ByteString, OptionVal }
  * Multiple instruments are automatically handled, however they MUST NOT overlap in their idenfitiers.
  *
  * Instances of `RemoteInstrument` are created from configuration. A new instance of RemoteInstrument
- * will be created for each encoder and decoder. It's only called from the stage, so if it dosn't
+ * will be created for each encoder and decoder. It's only called from the stage, so if it doesn't
  * delegate to any shared instance it doesn't have to be thread-safe.
  */
 abstract class RemoteInstrument {
   /**
    * Instrument identifier.
    *
-   * MUST be >=0 and <32.
+   * MUST be >=1 and <32.
    *
-   * Values between 0 and 7 are reserved for Akka internal use.
+   * Values between 1 and 7 are reserved for Akka internal use.
    */
   def identifier: Byte
 
   /**
-   * Called right before putting the message onto the wire.
-   * Parameters MAY be `null` (except `message`)!
-   *
-   * @return `metadata` rendered to be serialized into the remove envelope, or `null` if no metadata should be attached
+   * Should the serialization be timed? Otherwise times are always 0.
    */
-  def remoteMessageSent(recipient: ActorRef, message: Object, sender: ActorRef): ByteString
+  def serializationTimingEnabled: Boolean = false
 
   /**
-   * Called once a message (containing a metadata field designated for this instrument) has been deserialized from the wire.
+   * Called while serializing the message.
+   * Parameters MAY be `null` (except `message` and `buffer`)!
    */
-  def remoteMessageReceived(recipient: ActorRef, message: Object, sender: ActorRef, metadata: ByteString): Unit
-
-}
-
-object NoopRemoteInstrument extends RemoteInstrument {
-  override def identifier: Byte =
-    -1
+  def remoteWriteMetadata(recipient: ActorRef, message: Object, sender: ActorRef, buffer: ByteBuffer): Unit
 
-  override def remoteMessageSent(recipient: ActorRef, message: Object, sender: ActorRef): ByteString =
-    null
+  /**
+   * Called right before putting the message onto the wire.
+   * Parameters MAY be `null` (except `message` and `buffer`)!
+   */
+  def remoteMessageSent(recipient: ActorRef, message: Object, sender: ActorRef, size: Int, time: Long): Unit
```

Comment:
add some doc about what `size` and `time` are and cross reference to `serializationTimingEnabled`